core/src/main/java/org/elasticsearch/action/support/ToXContentToBytes.java

```diff
@@ -78,4 +80,20 @@ public final String toString() {
             return "{ \"error\" : \"" + ExceptionsHelper.detailedMessage(e) + "\"}";
         }
     }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+        if (other == null || !(other instanceof ToXContentToBytes)) {
+            return false;
+        }
+        return Objects.equals(defaultType, ((ToXContentToBytes) other).defaultType);
```

Comment:
We didn't serialize or use the defaultType of ToXContentToBytes so far, neither in the QueryBuilders nor in any other elements of the SearchSourceBuilder. I think it is safe to ignore this, because the only thing it does is to determine which type of xContent the object is written to when `buildAsBytes()` is called without an argument. It doesn't really change anything about how the object behaves on the shard.